### DISCUSSION TITLE
Cloudflare Adapter: Support for getPlatformProxy(), 'cache' and 'cf' object for local developments

### DISCUSSION THREAD

**Discussion by flikteoh** [QwikDev/qwik-evolution#196] [Proposals For Qwik]
### What is it about?

Support for Wrangler's getPlatformProxy() with cache and cf object

#### Problems you are trying to solve:
* Currently, Qwik implements Cloudflare Pages Adapter, but lacks the cf 'cache' api and 'cf' objects.
* Better feedback loops during local development
* Better DX

#### Goals you are trying to achieve:
* Update the current Cloudflare Pages Adapter / use `getPlatformProxy`
* Being able to iterate faster locally

#### Any other context or information you want to share:

While Qwik already has Cloudflare Pages Adapter implemented for Cloudflare Pages deployment, it lacks the abilities to work with their cache API and cf objects, locally.

We have to deploy to a development domain for every iterations (due to local dev breaks with `cache is not defined`), which consumes a lot of time for the builds ( > 1min per iteration).

This issue has [previously been brought up](https://github.com/QwikDev/qwik/issues/3345) for Cloudflare Pages, now with Cloudflare adding frameworks support for [Workers with Assets](https://developers.cloudflare.com/workers/frameworks/framework-guides/qwik/), which their [blog post](https://blog.cloudflare.com/blazing-fast-development-with-full-stack-frameworks-and-cloudflare/) mentioned that Qwik is also able to develop locally with services bindings.

Frameworks like [SvelteKit](https://svelte.dev/docs/kit/adapter-cloudflare#Options-platformProxy), [Astro](https://docs.astro.build/en/guides/integrations-guide/cloudflare/#platformproxy), [Remix](https://remix.run/docs/en/main/guides/vite#cloudflare-proxy) has implemented this via Wrangler's [getPlatformProxy](https://developers.cloudflare.com/workers/wrangler/api/#return-type-1) API.

It would be nice if this can also be supported in Qwik.

---

#### What do you propose?

Update Cloudflare Pages Adapter to include `cache` and `cf` object, via getPlatformProxy()

#### Code examples

https://github.com/withastro/adapters/blob/5391c024f1cd0b66b2db26c25e695cf76788b890/packages/cloudflare/src/index.ts#L196-L204

https://github.com/QwikDev/qwik/blob/7558018276e76441570c379351d7b3973b748fa4/packages/qwik-city/src/middleware/cloudflare-pages/index.ts#L144-L148

---

### Links / References

- https://blog.cloudflare.com/blazing-fast-development-with-full-stack-frameworks-and-cloudflare/
- https://developers.cloudflare.com/workers/wrangler/api/#return-type-1
- https://github.com/cloudflare/workers-sdk/blob/c1022d66c6536c9e8199ebffe73d2358aa4064ee/packages/create-cloudflare/templates-experimental/qwik/c3.ts#L110-L114
- https://svelte.dev/docs/kit/adapter-cloudflare#Options-platformProxy
- https://docs.astro.build/en/guides/integrations-guide/cloudflare/#platformproxy
- https://remix.run/docs/en/main/guides/vite#cloudflare-proxy

**Comment by shairez** [comment]
thanks for the proposal @flikteoh !

We're currently exploring moving to a nitro based approach, where most of these features are already implemented and maintained, so closing this in favor of that effort, feel free to re-open in the future if this isn't added with the nitro integration

thanks!

**Reply by flikteoh** [reply]
@shairez Thanks for the update! A bit lost here, what is a nitro based approach? Is there anywhere I can learn more about this?
